src/ir/context.rs

```diff
@@ -569,6 +569,24 @@ impl<'ctx> BindgenContext<'ctx> {
         self.used_template_parameters = Some(used_params);
     }
 
+    /// Return `true` if `item` uses the given `template_param`, `false`
+    /// otherwise.
+    ///
+    /// This method may only be called during the codegen phase, because the
+    /// template usage information is only computed as we enter the codegen
+    /// phase.
+    pub fn uses_template_parameter(&self, item: ItemId, template_param: ItemId) -> bool {
+        assert!(self.in_codegen_phase(),
```

Comment:
Perhaps we should move this to something like a `Phase` enum?

```rust
enum Phase {
    Parsing,
    ReplacementProcessing,
    Codegen,
}
```

With something like:

```rust
impl BindgenContext {
    pub fn advance_phase(&mut self, to: Phase) {
        assert!(to > self.current_phase);
        self.current_phase = to;
    }
}
```

And make more fine-grained assertions?

It's probably fine as a followup, or even an easy bug. Do you think it's worth filing?